packages/truffle-debugger/lib/data/reducers.js

```diff
@@ -59,8 +59,25 @@ function scopes(state = DEFAULT_SCOPES, action) {
   }
 }
 
+//a note on the following reducer: solidity assigns a unique AST ID to every
+//AST node among all the files being compiled together.  thus, it is, for now,
+//safe to identify user-defined types solely by their AST ID.  In the future,
+//once we eventually support having some files compiled separately from others,
+//this will become a bug you'll have to fix, and you'll have to fix it in the
+//decoder, too.  Sorry, future me! (or whoever's stuck doing this)
+
+function userDefinedTypes(state = [], action) {
```

Comment:
How do you feel about just calling this `types`? `userDefined` seems semi-extraneous, but I could go either way.

Reply:
Going to leave this one as-is, since it is indeed just the user-defined types.